src/pages/FavoritesPage/FavoritesPage.tsx

```diff
@@ -7,25 +7,28 @@ import { Alert, AlertTitle } from "@mui/material";
 import Grid from "@mui/material/Unstable_Grid2/Grid2";
 
 import styles from "./FavoritesPage.module.scss";
+import { FavoriteProduct } from "../types";
 
 const FavoritesPage = () => {
   const { favorites } = useFavoriteStore();
+  if (!favorites.length) {
+    return (
+      <Alert icon={<DiamondOutlined fontSize="inherit" />} severity="warning">
+        <AlertTitle>There aren't any favorite products</AlertTitle>
+        Please explore the <Link to="/">products page</Link>
+      </Alert>
+    );
+  }
 
   return (
     <div className={styles.favorites}>
       {favorites.length > 0 && (
         <Grid container>
-          {favorites.map((product) => {
+          {favorites.map((product: FavoriteProduct) => {
```

Comment:
Typescript get the type automatically here.
```suggestion
          {favorites.map((product) => {
```